src/platform/plugins/shared/expressions/common/expression_types/specs/datatable.ts

```diff
@@ -116,7 +116,6 @@ export function isSourceParamsESQL(obj: Record<string, unknown>): obj is SourceP
     obj &&
     typeof obj.indexPattern === 'string' &&
     typeof obj.sourceField === 'string' &&
-    typeof obj.operationType === 'string' &&
```

Comment:
why removing this check? `SourceParamsESQL.operationType` is a string, if we removed that because this `operationType` can have a different type than string, than there is a problem here

Reply:
The operationType is something being added by the UI Lens (the one that fallbacks to ES|QL is possible). When we have ES|QL mode where the user types ES|QL and then maps the columns to dimensions we don't have the operationType. So having this check here doesnt allow to move to the new check added.